clang/docs/SYCLMainPageforRFC.md

```diff
@@ -0,0 +1,68 @@
+
+We would like to lift these limitations, at least in the common case where the user uses clang for both the host and device compilation phases.  Our thought was to implement a different integration approach for this common case, which might assume that the host compiler understands clang attributes like __builtin_sycl_unique_stable_name and decorates the LLVM IR with some new attributes.  Our thinking is very early at this point.
+
+Even if we develop this new approach, though, we think it makes sense to keep the approach based on the integration headers because this will continue to be used for a third-party host compiler.
+
+### Location for the logic for generating the integration support
+
+Another open question is the location for the logic that generates the integration headers (or the new integration information alluded to above).  This logic is currently located in the Sema phase of the CFE, and the community has expressed concern about this.  We are considering two alternatives.  One option is to move the logic to the CodeGen phase of the CFE.  Another is to move the logic out of the CFE entirely and create a new LLVM IR pass instead.  If we do this, the new IR pass would likely run near the start of the pipeline, before other IR passes can transform the IR that is emitted by the CFE.  This option may be attractive especially if we end up supporting two integration methods because we could isolate each method in its own IR pass, and then enable one or the other pass depending on whether the user requests a third-party host compiler.
+
+### Dependency on the Khronos LLVM-SPIRV translator
+
+With our current approach, device code corresponding to the offload region is compiled into LLVM IR.  Various backend compilers (e.g., Intel Graphics Compiler), require the device code to be present in SPIR-V IR.  Hence, the LLVM IR needs to be translated into SPIR-V IR.
+
+This final translation to SPIR-V currently relies on the Khronos SPIRV-LLVM translator, which is a project external to the LLVM compiler infrastructure.  As a result, end users must ensure that the translator is available in their environment, so that the clang front-end driver is able to find it during compilation for SYCL.  We would like to eliminate this dependency in order to make it easier for people to use clang for SYCL.
+
+A SPIR-V backend is currently under development inside the LLVM compiler infrastructure.  Once deployed, this backend can replace the current translator in the SYCL compilation flow.
```

Comment:
```suggestion
With our current approach, device code corresponding to offload region in SYCL source file is compiled and linked into LLVM IR. Various backend compilers (Intel Graphics Compiler, etc), require the device code to be present in SPIR-V IR. Hence, we need a compiler flow to generate SPIR-V IR from LLVM IR during the post-link stage.
This final translation to SPIR-V IR currently relies on the Khronos SPIRV-LLVM translator, which is a project external to LLVM compiler infrastructure.  As a result, end users must ensure that the translator is available in their environment, so that the clang driver is able to find it during post-link stage of compilation for SYCL.  We would like to eliminate this dependency in order to make it easier for people to use clang for SYCL.
A SPIR-V backend is currently under development inside LLVM compiler infrastructure. Once deployed, this backend can replace the current translator in the post-link stage of SYCL compilation flow.
```

Reply:
Thanks, @asudarsa.  Yes, this looks better.